source/MaterialXFormat/XmlExport.cpp

```diff
@@ -50,19 +50,33 @@ void exportToXmlStream(DocumentPtr doc, std::ostream& stream, const XmlExportOpt
 void exportToXmlFile(DocumentPtr doc, const FilePath& filename, const XmlExportOptions* exportOptions)
 {
     mergeLooks(doc, exportOptions);
-    if (exportOptions && exportOptions->flattenFilenames)
+    if (exportOptions)
     {
-        flattenFilenames(doc, exportOptions->resolvedTexturePath, exportOptions->stringResolver);
+        for (ResolverPtr resolver : exportOptions->resolvers)
```

Comment:
Must be called after flatten since that converts from relative to absolute path.
Abs paths are required for colorspace resolving.